test/integration/directory_artifact_test.js

```diff
@@ -79,4 +83,81 @@ suite('Directory artifact', () => {
       'Size of uploaded contents of test.html does not match original.'
     );
   });
+
+  test('retry directory upload', async () => {
+    let ARTIFACT_COUNT = 3;
+
+    let worker = new TestWorker(DockerWorker);
+    await worker.launch();
+
+    var count = 0;
+
+    let rejectConnection = function() {
+      // Delay iptables rejection to catch
+      // in the middle of the upload
+      setTimeout(function() {
+        iptables.reject({
```

Comment:
so now that I've found out that at least one person will run docker tests locally outside of a vagrant environment, this makes me very scared that this doesn't get cleaned up after a test and all secure connections are then rejected after running the test.  Perhaps there is a way we can find out if this is running in our vagrant environment or automation and run it? I thought that this would be an easy use of iptables magic, but it seems the downsides are starting to pile up.